extensions/amp-ad/0.1/amp-ad-3p-impl.js

```diff
@@ -127,6 +127,21 @@ export function isPositionFixed(el, win) {
   return hasFixedAncestor;
 }
 
+/**
+ * @param {!Element} el
+ * @param {!Window} win
+ * @return {string|null} a string of the name of valid containers of the ad.
+ * This is called before creating iframe for this ad.
+ */
+export function getAdContainer(el) {
+  do {
+    if (POSITION_FIXED_TAG_WHITELIST[el.tagName]) {
+      return el.tagName;
+    }
+    el = el.parentNode;
+  } while (el && el.tagName != 'BODY');
```

Comment:
1) This method doesn't return `null` at the end of the loop.
2) We might start the loop from `el.parent` because the method name implies the check is against container, not the element itself.